content/docs/user-guide/programming/_index.md

```diff
@@ -43,6 +43,20 @@ Examples of high-level products that use these frameworks are:
 * **GUI tools**: Tools with a graphical user experience are used for developing O3DE projects, such as **O3DE Editor**, **Asset Processor**, and **Material Editor**.
 
 
+### O3DE directories
+
+If you're interested in developing O3DE, creating components and Gems, or programming in your project, it's helpful to be familiar with the following subdirectories in O3DE's source:
+
+  * `cmake`: Contains configuration, download, and build scripts for O3DE.
+
+  * `Code`: Contains the C++ code and headers that are used to build O3DE and provide its APIs. APIs are organized by libraries, each of which consists of a well-defined feature set. Library headers offer virtual interfaces that you can provide implementations of to connect your code to the relevant O3DE system or feature.
+
+  * `Code/Framework`: Contains all of the source code and headers used by O3DE's core modules.
+
+  * `Gems`: Contains the source and build files for the core Gems. Each Gem has its own subdirectory. When you create a new Gem, you can add the code and build sources here, and then enable the Gem in your project configuration. It's possible for a Gem to contain other Gems. The Atom Gem, for example, contains several Gems providing various tools, libraries, interfaces, and utilities for Atom Renderer.
```

Comment:
I wouldn't call these Core Gems, so much as these are Gems that come with the Engine(i.e Engine Gems).

Some gems will be moved out of the o3de repo to other repo.

Reply:
Noted, we can refer to them as engine Gems as that is more accurate in the long term.

Reply:
@lumberyard-employee-dm Noted, I think we can refer to them here as "Gems that come with the engine". A bit long, but "engine Gems" makes me think that the Gems make up the core engine, rather than add on top of the engine. 
@willihay Thoughts?

Reply:
Since the default location for user-created Gems is in the `<user>/Gems` directory and not here, I'll also remove the sentence "When you create a new Gem, you can add the code and build sources here, and then enable the Gem in your project configuration." This sentence is true if users create a new Gem here, but may just add confusion in this context.

Reply:
Yeah, I agree. If a reader didn't pick up on the emphasis that you *can* create Gems here, but you aren't required to, then they might think you need to create them there.

On the topic of "core Gems", we have a terminology problem, because I just realized we have a topic in the section on Gems in the User Guide called [Core Gems](https://www.o3de.org/docs/user-guide/gems/core-gems/) (which do **not** include _all_ the Gems in the engine source), not to mention the "O3DE Core Gem" (LmbrCentral). So I have to agree with dm that referring to all the Gems that come with the engine as "core Gems" is a bad idea.

In the context here, I suggest referring to them for what they are, "Gems included with the engine", until we come up with a better/standard term.

Reply:
I am for the term "Gems included with the engine", until we have a better term. Especially since there are talks of moving some gems out of the repo.